enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,543 @@
+types without detriment.
+
+Spread placement groups have a limited capacity. You can only launch seven instances within a Spread placement group
+within a single availability zone.
+
+###### Limitations of Spread Placement Groups
+
+- Spread placement groups are not supported for AWS Dedicated Instances.
+
+##### Partition Placement Groups
+
+A Partition placement group divides your instances into logical segments called partitions.
+Each partition is assigned a set of racks within the Availability Zone, each rack in turn having its own network and
+power source.
+No rack within the AZ will appear in more than one partition.
+
+Users may either create instances and have them spread automatically across partitions or they may allocate instances
+to a particular partition.
+Partitions are identified numerically, indexed from 1.
+
+Partition placement groups are recommended for deploying large distributed and replicated workloads across distinct
+racks.
+
+Partition placement groups are limited to seven partitions per Availability Zone.
+
+When using a Partition placement group, if users wish to control the placement of their workloads within Partitions,
+they will need to create multiple MachineSets and assign each one to a partition.
+Once they have assigned different MachineSets to partitions, they will be able to use Node affinity to assign the
+workloads to the Nodes created by the Machines within the MachineSet by selecting based on the MachineSets unique
+labels.
+
+For example, across three MachineSets, the Placement would vary only by the partition number:
+
+```yaml
+providerSpec:
+  value:
+    placement:
+      region: us-east-1
+      availabilityZone: us-east-1a
+      groupName: my-partition-placement-group
+      groupType: Partition
+      partitionCount: 3
+      partitionNumber: 1 | 2 | 3 # This value will vary across the three MachineSets, either 1, 2 or 3
+```
+
+###### Limitations of Partition Placement Groups
+
+- A partition placement group with AWS Dedicated Instances can have a maximum of two partitions.
+
+#### Creation of Placement Groups
+
+To enable users to get started with Placement Groups, we will allow users to specify the name of a placement group via
+the new `placement.groupName` field on their MachineSet, which will then be used for the placement of new Machines
+created by the Machineset.
+If the placement group does not exist, we will create it based on the configuration provided by the user.
+
+We will allow the user to specify the type of the placement group to create, either Cluster, Spread or Partition, as
+well as, in the case of Partition, the number of partitions to be created.
+
+Should the user not define the type, we will return an error and place the Machine into a Failed state. Due to the
+differences in behaviour between the different placement group strategies, we do not believe there is a sensible default
+choice and as such, the group type will be required when creating a placement group.
```

Comment:
> Use an admission plugin or webhook to reject the machineset if it specifies groupName xor groupType.

We already have webhooks set up so this was my intended plan for this validation

> Admit such a machineset and then report the error in status, as described here.

With this logic in the Machine controller in case the webhook somehow failed

All of the restrictions imposed by the kubebuilder tags are intended to be added to our existing admission webhooks, this hasn't been implemented yet so I will make sure to add a card to track that